data/scripts/dependency-resolver.py

```diff
@@ -540,7 +540,7 @@ def log_environment(context):
 
     args = parser.parse_args()
 
-    if args.compiler != default_compiler and not which(args.compiler):
+    if args.compiler != default_compiler and not which(args.compiler.split(" ")[0]):
```

Comment:
Indeed, using split() should cover more possibilities. I will merge with this thanks